presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -721,16 +746,24 @@ private void revokeRole(String role, String granteeName, PrincipalType granteeTy
         }
     }
 
-    private List<String> getPrestoViews(String databaseName)
+    private List<String> doGetTablesWithParameter(String databaseName, String parameterKey, String parameterValue)
             throws TException
     {
+        /*
+         * The parameter value is restricted to have only alphanumeric characters so that it's safe to be used against HMS.
+         * When using with a LIKE operator, the HMS may want the parameter value to follow a Java regex pattern or a SQL
+         * pattern. And it's hard to predict the HMS's behavior from outside.
+         */
+        if (!TABLE_PARAMETER_VALUE_PATTERN.matcher(parameterValue).matches()) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, "The parameter value has one or more non-alphanumeric characters");
+        }
         /*
          * Thrift call `get_table_names_by_filter` may be translated by Metastore to a SQL query against Metastore database.
          * Hive 2.3 on some databases uses CLOB for table parameter value column and some databases disallow `=` predicate over
          * CLOB values. At the same time, they allow `LIKE` predicates over them.
          */
-        String filterWithEquals = HIVE_FILTER_FIELD_PARAMS + PRESTO_VIEW_FLAG + " = \"true\"";
-        String filterWithLike = HIVE_FILTER_FIELD_PARAMS + PRESTO_VIEW_FLAG + " LIKE \"true\"";
+        String filterWithEquals = HIVE_FILTER_FIELD_PARAMS + parameterKey + " = \"" + parameterValue + "\"";
```

Comment:
`parameterKey` also needs to be validated.
I think we can assume `^[a-zA-Z]+$`.